scenarios/test_codegen/test/lib_tests/EntityHistory_test.res

```diff
@@ -3,46 +3,51 @@ open RescriptMocha
 //unsafe polymorphic toString binding for any type
 @send external toStringUnsafe: 'a => string = "toString"
 
-type testEntity = {
-  id: string,
-  fieldA: int,
-  fieldB: option<string>,
-}
-
-let testEntitySchema: S.t<testEntity> = S.schema(s => {
-  id: s.matches(S.string),
-  fieldA: s.matches(S.int),
-  fieldB: s.matches(S.option(S.string)),
-})
-
-let testEntityRowsSchema = S.array(testEntitySchema)
+module TestEntity = {
+  type t = {
+    id: string,
+    fieldA: int,
+    fieldB: option<string>,
+  }
+
+  let name = "TestEntity"->(Utils.magic: string => Enums.EntityType.t)
```

Comment:
This is dangerous if it were to match on the entity type but right now the code doesn't do that.

Reply:
In the future I'd actually want to:
1. Remove the enum for entity type
2. Change entityMod to simply be a record type

Reply:
Sounds good 💪🏼, I have some ideas as well. Eg. I'm keen to derive the table from the rescript schema. And do some runtime validations there to start making it more pluggable as a library without needing a gql schema file.

Reply:
This is exactly how I visioned it myself 👍